main/src/modules/Jvm.scala

```diff
@@ -283,10 +283,13 @@ object Jvm {
 
     Assembly.groupAssemblyEntries(inputPaths, assemblyRules).view
       .foreach {
-        case (mapping, AppendEntry(entries)) =>
+        case (mapping, AppendEntry(entries, separator)) =>
           val path = zipFs.getPath(mapping).toAbsolutePath
+          val separated = entries.flatMap { e =>
+            List(e, JarFileEntry(e.mapping, () => new ByteArrayInputStream(separator.getBytes)))
+          }
```

Comment:
Looks like the separator is also added after the last entry. I'd rather keep a naive separator semantic in terms of "separate" and avoid adding it to the end.

Imagine the case where exactly one "reference.conf" is found. The above impl. would change the file contents (and add a separator) instead of just keeping it as-is.

Reply:
Hm, yes I agree with you